File: backend/Proppy.API/Domain/Models/Employee.cs
using System.Collections.Generic;
using System;

namespace Proppy.API.Domain.Models
{
    public class Employee
    {
        public int ID {get; set;}
        public string Name {get; set;}
        public string Phone_No {get; set;}
        public string Email{get; set;}
        public string Gender {get; set;}
        public Position Position {get; set;}
        public string Position_Code {get; set;}
        public DateTime DOB {get; set;} 
        public int Salary {get; set;}
        public string Remarks {get; set;}
        public string Photo {get; set;}
    }
}

File: app/src/models/Employee.ts
import { ErrorObject } from "./Error";
import { Position } from "./Position";

export interface EmployeeBody {
  name: string;
  email: string;
  phoneNo: string;
  gender: string;
  positionCode: string;
  salary: number;
  photo: string;
  dob: Date;
  remarks: string;
}

export interface EmployeeResponse {
  id: number;
  name: string;
  email: string;
  phone_No: string;
  gender: string;
  position: Position;
  salary: number;
  photo: string;
  dob: Date;
  remarks: string;
}

export interface EmployeeQueryResponse {
  totalItems: number;
  items: EmployeeResponse[];
}

export interface EmployeeListStore {
  isLoading: boolean;
  isRetrieved: boolean;
  error: ErrorObject;
  employees: EmployeeQueryResponse;
}

export interface EmployeeDetailStore {
  isLoading: boolean;
  isRetrieved: boolean;
  error: ErrorObject;
  employees: Employee[];
}

export interface EmployeeCreateStore {
  isLoading: boolean;
  error: ErrorObject;
  employee: EmployeeBody;
}
